File: Core/Models/Word.swift
//
//  Word.swift
//  My Dictionary
//
//  Created by Aleksandr Riakhin on 3/8/25.
//

import Foundation

struct Word: Identifiable, Hashable {
    let word: String
    var definition: String
    var partOfSpeech: PartOfSpeech
    var phonetic: String
    let id: String
    let timestamp: Date
    var examples: [String]
    var isFavorite: Bool

    init(
        word: String,
        definition: String,
        partOfSpeech: PartOfSpeech,
        phonetic: String,
        id: String,
        timestamp: Date,
        examples: [String],
        isFavorite: Bool
    ) {
        self.word = word
        self.definition = definition
        self.partOfSpeech = partOfSpeech
        self.phonetic = phonetic
        self.id = id
        self.timestamp = timestamp
        self.isFavorite = isFavorite
        self.examples = examples
    }
}

File: MyDictionaryAndroid/app/src/main/java/com/dor/mydictionary/core/Word.kt
package com.dor.mydictionary.core

import com.dor.mydictionary.core.PartOfSpeech
import java.util.Date

data class Word(
    val wordItself: String,
    val definition: String,
    val partOfSpeech: PartOfSpeech,
    val phonetic: String?,
    val id: String,
    val timestamp: Date,
    val examples: List<String>,
    val isFavorite: Boolean,
    val difficultyScore: Int
) {
    val difficultyLevel: Difficulty
        get() = Difficulty.fromScore(difficultyScore)
    
    val difficultyLabel: String
        get() = difficultyLevel.displayName
    
    val shouldShowDifficultyLabel: Boolean
        get() = difficultyScore > 0
}